studiopress/pattern-manager
# [GF-3828] Detect Local Environment, Add Dismissible Notice

## Tasks


### Summary of changes


Users of Pattern Manager should be working with a local development environment. This PR adds a simple check for a local environment, then displays a dismissible notice if a non-local environment is detected and the user has not previously dismissed the notice.

The notice displays an external link to download Local WP: https://localwp.com/?modal=download.

### Related Issues



### How to test

Using Local WP
1. Checkout the branch
2. Navigate to the main patterns view (`/wp-admin/admin.php?page=pattern-manager`)
3. The notice should **_NOT_** display since you are on a local environment

Using a WPE-Hosted Site
1. Checkout the branch
2. Create or use an existing site in my_wpe (https://my.wpengine.com/bulk_sites)
3. Install and activate the attached, zipped plugin: [pattern-manager.0.3.0.zip](https://github.com/studiopress/pattern-manager/files/12043046/pattern-manager.0.3.0.zip)
4. Navigate to the main patterns view (`/wp-admin/admin.php?page=pattern-manager`)
5. The notice should display
6. Dismiss the notice and refresh the page
7. The notice should no longer be displayed

### Notes & Screenshots


![pm_local-error](https://github.com/studiopress/pattern-manager/assets/108079556/15750067-fdb1-451e-8609-51126487592a)
